salt/states/rabbitmq_vhost.py

```diff
@@ -70,24 +70,22 @@ def present(name):
 
     vhost_exists = __salt__['rabbitmq.vhost_exists'](name)
 
+    if vhost_exists:
+        ret['comment'] = 'VHost {0} already exists'.format(name)
+        return ret
+
     if __opts__['test']:
         ret['result'] = None
-        if vhost_exists:
-            ret['comment'] = 'VHost {0} already exists'.format(name)
-        else:
-            ret['comment'] = 'Creating VHost {0}'.format(name)
-
-    else:
-        if vhost_exists:
-            ret['comment'] = 'VHost {0} already exists'.format(name)
-        else:
-            result = __salt__['rabbitmq.add_vhost'](name)
-            if 'Error' in result:
-                ret['result'] = False
-                ret['comment'] = result['Error']
-            elif 'Added' in result:
-                ret['comment'] = result['Added']
-                ret['changes'] = {'old': '', 'new': name}
+        ret['comment'] = 'Creating VHost {0}'.format(name)
+        return ret
+
+    result = __salt__['rabbitmq.add_vhost'](name)
+    if 'Error' in result:
+        ret['result'] = False
+        ret['comment'] = result['Error']
+    elif 'Added' in result:
+        ret['comment'] = result['Added']
+        ret['changes'] = {'old': '', 'new': name}
```

Comment:
This looks great! One more tiny thing, then, since we're here. The `changes` dictionary should also show up on a state run when running `test=true`. Can you make sure the `changes` dict gets placed in the `if __opts__['test']` section, too?